views/site_files/text_editor.erb

```diff
@@ -111,10 +112,18 @@
   <script src="/js/chat.js"></script>
 <% end %>
 -->
+
+<script src="/js/preview.js"></script>
+
 <div class="row editor">
   <div class="col left-col" style="display: <% current_site.supporter? ? 'none' : 'block' %>; margin-left: 20px;">
     <div id="editor"><h3>Loading...</h3></div>
   </div>
+  <div id="preview" class="col right-col chat-container" style="display:none;">
+    <div class="resize-handle"></div>
+    <iframe id="previewFrame" src="<%= current_site.uri @filename %>" width="100%" height="100%" style="background: white"></iframe>
+  </div>
```

Comment:
This is a rare scenario and probably a nitpick but if they're editing an HTML file that automatically redirects them to another, the code works so that the website is loaded before you even enable the preview feature. Because of this, the user would be redirected before they get to see the file they're trying to preview.

Solutions for this would be to either load the website only if the preview bar is visible, or to warn the user that the preview has been redirected to a different file.

Reply:
This would cause a performance overhead for every single webpage that doesn't have this hypothetical "automatic redirect" because it would be forced to refresh the page each time they open the preview.

Reply:
Ohh I see...